photon-ml/src/main/scala/com/linkedin/photon/ml/avro/data/ScoreProcessingUtils.scala

```diff
@@ -51,9 +57,19 @@ object ScoreProcessingUtils {
    * @param outputDir The given output directory
    */
   protected[ml] def saveScoredItemsToHDFS(scoredItems: RDD[ScoredItem], modelId: String, outputDir: String): Unit = {
-    val scoringResultAvros = scoredItems.map { case ScoredItem(uid, predictionScore) =>
-      val avroFile = ScoringResultAvro.newBuilder().setUid(uid).setModelId(modelId).setPredictionScore(predictionScore)
-      avroFile.build()
+    val scoringResultAvros = scoredItems.map { case ScoredItem(predictionScore, uidOpt, labelOpt, ids) =>
+      val metaDataMap = collection.mutable.Map(ids.toMap[CharSequence, CharSequence].toSeq: _*).asJava
+      val builder = ScoringResultAvro.newBuilder()
+      builder.setPredictionScore(predictionScore)
+      builder.setModelId(modelId)
+      if (uidOpt.isDefined) {
+        builder.setUid(uidOpt.get)
+      }
+      if (labelOpt.isDefined) {
+        builder.setLabel(labelOpt.get)
+      }
```

Comment:
I believe the recommended syntax for these Option checks would be:
```
uidOpt.foreach(builder.setUid(_))
labelOpt.foreach(builder.setLabel(_))
```
http://blog.tmorris.net/posts/scalaoption-cheat-sheet/